google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/TopicAdminSettings.java

```diff
@@ -106,131 +78,74 @@
 @Generated("by GAPIC v0.0.5")
 @BetaApi
 public class TopicAdminSettings extends ClientSettings<TopicAdminSettings> {
-  /** The default scopes of the service. */
-  private static final ImmutableList<String> DEFAULT_SERVICE_SCOPES =
-      ImmutableList.<String>builder()
-          .add("https://www.googleapis.com/auth/cloud-platform")
-          .add("https://www.googleapis.com/auth/pubsub")
-          .build();
-
-  private final UnaryCallSettings<Topic, Topic> createTopicSettings;
-  private final UnaryCallSettings<UpdateTopicRequest, Topic> updateTopicSettings;
-  private final BatchingCallSettings<PublishRequest, PublishResponse> publishSettings;
-  private final UnaryCallSettings<GetTopicRequest, Topic> getTopicSettings;
-  private final PagedCallSettings<ListTopicsRequest, ListTopicsResponse, ListTopicsPagedResponse>
-      listTopicsSettings;
-  private final PagedCallSettings<
-          ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse,
-          ListTopicSubscriptionsPagedResponse>
-      listTopicSubscriptionsSettings;
-  private final UnaryCallSettings<DeleteTopicRequest, Empty> deleteTopicSettings;
-  private final UnaryCallSettings<SetIamPolicyRequest, Policy> setIamPolicySettings;
-  private final UnaryCallSettings<GetIamPolicyRequest, Policy> getIamPolicySettings;
-  private final UnaryCallSettings<TestIamPermissionsRequest, TestIamPermissionsResponse>
-      testIamPermissionsSettings;
+  private final PublisherStubSettings stubSettings;
+
+  public static final TopicAdminSettings create(PublisherStubSettings stub) throws IOException {
+    return new TopicAdminSettings.Builder(stub).build();
+  }
+
+  public PublisherStubSettings getStubSettings() {
+    return stubSettings;
+  }
 
   /** Returns the object with the settings used for calls to createTopic. */
   public UnaryCallSettings<Topic, Topic> createTopicSettings() {
-    return createTopicSettings;
+    return stubSettings.createTopicSettings();
   }
 
   /** Returns the object with the settings used for calls to updateTopic. */
   public UnaryCallSettings<UpdateTopicRequest, Topic> updateTopicSettings() {
-    return updateTopicSettings;
+    return stubSettings.updateTopicSettings();
   }
 
   /** Returns the object with the settings used for calls to publish. */
   public BatchingCallSettings<PublishRequest, PublishResponse> publishSettings() {
-    return publishSettings;
+    return stubSettings.publishSettings();
```

Comment:
Any surface for the `publish` API should be hidden on `TopicAdminSettings`. In the code generator, this should be driven by the surface_treatments.visibility config ( https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub_gapic.yaml#L515 ).

Reply:
Does that mean that this method should be package-private, as specified in that gapic config?

Reply:
Yes.

Reply:
Ok, made this method and the builder method package private.